File: perl/prev-year-data-process.pl
#!/usr/bin/perl
use JSON;
use IO::Handle;

sub createDataset {
    open(my $json_file, '<', '../pan/pan-demo-smalll.jsonl') or die $!;
    open(my $truth_file, '<', '../pan/pan-demo-small-truth.jsonl') or die $!;
    my @truth_list = <$truth_file>;
    open(my $cases, '>>', 'truthvalues.txt') or die $!;
    my @json_list = <$json_file>;
    my $i = 0;
    my $pair_num = 0;
    foreach my $entry (@json_list) {
        my $js = $entry;
        my $result = decode_json($js);
        my @textPair = @{$result->{'pair'}};
        open(my $f1, '>', "../data/$i.txt") or die $!;
        print {$f1} "$textPair[0]";
        close($f1);
        $i++;
        open(my $f2, '>', "../data/$i.txt") or die $!;
        print {$f2} "$textPair[1]";
        close($f2);
        $i++;
        my $truthjs = $truth_list[$pair_num];
        $pair_num++;
        my $tresult = decode_json($truthjs);
        my $ans = 0;
        if ($tresult->{'same'} == 1) {
            $ans = 1;
        }
        print {$cases} "$i-2.txt $i-1.txt $ans\n";
    }
    close($cases);
}

createDataset();

File: prev-year-data-process.py
import json
import io
def createDataset():
    with open('pan/pan-demo-smalll.jsonl', 'r') as json_file:
        with open('pan/pan-demo-small-truth.jsonl', 'r') as truth_file:
            truth_list = list(truth_file)
            with open("truthvalues.txt", "a") as cases:
                json_list = list(json_file) 
                i=0
                pair_num = 0
                for entry in json_list:
                    #something like {"fandom":"harrypotter, "pair": ["I love coding.", "Coding is fun!"]}

                    js = entry
                    #parse as JSON
                    result = json.loads(js)
                    #textPair is assigned the value of result['pair'], which is ["I love coding.", "Coding is fun!"]
                    textPair = result['pair']
                    #first file
                    with io.open("data/" + str(i) + ".txt", 'w', encoding='utf-8') as f1:
                        f1.write(textPair[0])
                        f1.close()
                    i += 1
                    #second file
                    with io.open("data/" + str(i) + ".txt", 'w',encoding='utf-8') as f2:
                        f2.write(textPair[1])
                        f2.close()
                    i += 1
                    truthjs = truth_list[pair_num]
                    pair_num+=1
                    tresult = json.loads(truthjs)
                    ans = 0
                    if (tresult['same'] == True):
                        ans = 1
                    cases.write(str(i-2) + ".txt " + str(i-1) + ".txt " + str(ans) + "\n")
    cases.close()
createDataset()
